### PR TITLE
[4.0] setting icon now accessible

### PR DESCRIPTION
**Pull request by bahl24**
Pull Request for Issue #24325  .

### Summary of Changes
Added empty href attribute for settings icon & unpublish option in dropdown

### Testing Instructions
Go to Control Panel.
Tab through whole page using keyboard.

### Expected result
Setting Icon & unpublish option of all Administrator Modules (Sample Data, Logged-in Users etc.) can be selected for keyboard only users via TAB.

### Actual result
Not able to select Setting icon & unpublish option through keyboard.

### Documentation Changes Required
None

### CODE REVIEW

**Comment by wilsonge** [comment]
I think this is another one of these places it should be a button instead of a link - because it's just opening the dropdown

**Comment by bahl24** [comment]
@wilsonge Actually, the dropdown functionality was added recently in https://github.com/joomla/joomla-cms/pull/23934. In alpha7, it was a link with href redirecting to edit, href tag was removed in alpha8, which results in a11y issue.
If you want I can change it to a button.

**Comment by brianteeman** [comment]
it should be handled exactly the same as any dropdown menu item

**Comment by bahl24** [comment]
Other dropdown menus such as sidebar-nav & User menu use `<a href="#"... ` tag only and are not buttons.
Screenshot from 2019-03-26 06-47-09
Screenshot from 2019-03-26 06-51-29
@wilsonge So should we change all dropdowns to buttons?

**Comment by wilsonge** [comment]
@zwiastunsw @brianteeman i would assume these should be buttons. But before I give the word could one of you two confirm please

**Comment by zwiastunsw** [comment]
It should be a button, not a link. Setting a with href="#" attribute is incorrect.

**Comment by bahl24** [comment]
No problem, let me change them to button and not links

**Comment by bahl24** [comment]
@wilsonge Kindly check the changes

**Comment by brianteeman** [comment]
lgtm

**Comment by bahl24** [comment]
@wilsonge @zwiastunsw Kindly test

File: administrator/templates/atum/html/layouts/chromes/well.php [hunk 1]
```diff
@@ -44,14 +44,14 @@
...
+					<button type="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" id="dropdownMenuButton-<?php echo $id; ?>">
 						<span class="fa fa-cog"><span class="sr-only">
 							<?php echo Text::_('JACTION_EDIT') . ' ' . $module->title; ?>
 						</span></span>
-					</a>
+					</button>
 					<div class="dropdown-menu dropdown-menu-' . $dropdownPosition . '" aria-labelledby="dropdownMenuButton-<?php echo $id; ?>">
 						<a class="dropdown-item" href="<?php echo $url; ?>"><?php echo Text::_('JACTION_EDIT'); ?></a>
-						<a class="dropdown-item unpublish-module" data-module-id="<?php echo $id; ?>"><?php echo Text::_('JACTION_UNPUBLISH'); ?></a>
+						<a href="#" class="dropdown-item unpublish-module" data-module-id="<?php echo $id; ?>"><?php echo Text::_('JACTION_UNPUBLISH'); ?></a>
```
Comment (brianteeman) [inline administrator/templates/atum/html/layouts/chromes/well.php]: this still is not accessible. unlike the edit link above this is not a link to somewhere else but it is a direct action ie it is a button and should be coded as a button

Comment (bahl24) [reply administrator/templates/atum/html/layouts/chromes/well.php]: Ok, will change it.

Comment (bahl24) [reply administrator/templates/atum/html/layouts/chromes/well.php]: Done

File: administrator/templates/atum/html/layouts/chromes/well.php [hunk 2]
```diff
@@ -44,14 +44,14 @@
...
+					<button type="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" id="dropdownMenuButton-<?php echo $id; ?>">
 						<span class="fa fa-cog"><span class="sr-only">
 							<?php echo Text::_('JACTION_EDIT') . ' ' . $module->title; ?>
 						</span></span>
-					</a>
+					</button>
 					<div class="dropdown-menu dropdown-menu-' . $dropdownPosition . '" aria-labelledby="dropdownMenuButton-<?php echo $id; ?>">
 						<a class="dropdown-item" href="<?php echo $url; ?>"><?php echo Text::_('JACTION_EDIT'); ?></a>
-						<a class="dropdown-item unpublish-module" data-module-id="<?php echo $id; ?>"><?php echo Text::_('JACTION_UNPUBLISH'); ?></a>
+						<button class="dropdown-item unpublish-module" data-module-id="<?php echo $id; ?>"><?php echo Text::_('JACTION_UNPUBLISH'); ?></button>
```
Comment (wilsonge) [inline administrator/templates/atum/html/layouts/chromes/well.php]: Button needs a type attribute

Comment (bahl24) [reply administrator/templates/atum/html/layouts/chromes/well.php]: Done.

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**